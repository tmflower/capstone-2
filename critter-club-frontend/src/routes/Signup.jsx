import { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { Paper, FormControl, TextField, Button, Stack, Box, Alert } from '@mui/material';
 
/** Signup renders a form with Inputs for username, password, and access code from parent
 * Enables a user to sign up for a new account  */

export function Signup({ signup, alert }) {

    // set initial form fields to blank
    const initial_state = {
        username: '',
        password: '',
        accessCode: ''
    }
    
    // control masking and unmasking of password field
    const [passwordShowing, setPasswordShowing] = useState(false);
    const toggle = () => {setPasswordShowing(!passwordShowing)}

    const [formData, setFormData] = useState(initial_state);

    const { username, password, accessCode } = formData;

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(formData => ({ ...formData, [name]: value}));
    }

    // when user submits form, call signup function passed from App.js
    async function handleSubmit (evt) {
        evt.preventDefault();
        const newUser = { username, password, accessCode };        
        await signup(newUser);  
        localStorage.setItem("username", username);
        setFormData(initial_state);       
    }

    // variable and function to control display of alerts
    // const [alertShowing, setAlertShowing] = useState(true);
    // const closeAlert = () => setAlertShowing(!alertShowing);

    return (
        <Paper
        elevation={8}
        sx={{ padding: 3 }}>
            {/* {alertShowing && */}
            {alert.message.length ? 
            <Alert variant="filled" severity={alert.severity}>{alert.message}</Alert> : null}

            {/* <Alert severity={alert.severity} onClose={() => {closeAlert()}}>{alert.message}</Alert> : null} */}
            <h1>Join the Critter Club!</h1>
            <p>You'll need the access code from your parent's account to sign up.</p>
            <p>Don't have an access code yet? Grab your nearest parent and ask them to hop on over to the <NavLink to="/parent">Parent Page</NavLink>.</p>
            <form>
                
                <FormControl sx={{ width: '25ch', m: 2 }}>
                <Stack spacing={2}>
                
                {/* <InputLabel htmlFor="username">Username<small> (don't use your real name!)</small>:
                <Input 
                    type="text" 
                    name="username" 
                    value={username} 
                    id="username" 
                    onChange={handleChange}>
                </Input>
                </InputLabel> */}
                
                {/* <InputLabel htmlFor="username">Username<small> (don't use your real name!)</small>: */}
                <TextField 
                    type="text" 
                    name="username" 
                    value={username} 
                    id="username" 
                    label="Username"
                    onChange={handleChange}>
                {/* </InputLabel> */}
                </TextField>
                
                {/* <InputLabel htmlFor="password">Password: */}

                <Box>
                    <TextField 
                        type={passwordShowing ? "text" : "password"} 
                        name="password" 
                        value={password} 
                        id="password" 
                        label="Password"
                        onChange={handleChange}>
                    </TextField>

                        {!passwordShowing ?
                        <span className="material-symbols-outlined" onClick={toggle}>visibility</span>
                        :
                        <span className="material-symbols-outlined" onClick={toggle}>visibility_off</span>}
                </Box>


                {/* </InputLabel> */}

                {/* <InputLabel htmlFor="accessCode">Access code: */}
                <TextField                
                    type="text" 
                    name="accessCode" 
                    value={accessCode} 
                    id="accessCode"
                    label="Access code" 
                    onChange={handleChange}>
                </TextField>
                {/* </InputLabel> */}

                <Button id="signup-button" onClick={handleSubmit}>Submit</Button> 
                </Stack> 
                </FormControl> 
                           
            </form>
            
        </Paper>
    )
}