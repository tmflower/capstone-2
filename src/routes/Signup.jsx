import { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { Paper, FormControl, TextField, Button, Stack, Box, Alert } from '@mui/material';
 
/** Signup renders a form with Inputs for username, password, and access code from parent
 * Enables a user to sign up for a new account  */

export function Signup({ signup, alert }) {

    // set initial form fields to blank
    const initial_state = {
        username: '',
        password: '',
        accessCode: ''
    }

    // control masking and unmasking of password field
    const [passwordShowing, setPasswordShowing] = useState(false);
    const toggle = () => {setPasswordShowing(!passwordShowing)}

    const [formData, setFormData] = useState(initial_state);

    const { username, password, accessCode } = formData;

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(formData => ({ ...formData, [name]: value}));
    }

    // when user submits form, call signup function passed from App.js
    async function handleSubmit (evt) {
        evt.preventDefault();
        const newUser = { username, password, accessCode };        
        await signup(newUser);
        localStorage.setItem("username", username);
        setFormData(initial_state);       
    }

    return (
        <Paper
        elevation={8}
        sx={{ padding: 3 }}>            
            {alert.message.length ? 
            <Alert 
                variant="filled" 
                severity={alert.severity}>{alert.message}
            </Alert> 
            : null}            
            <h1>Join the Critter Club!</h1>
            <p>You'll need the access code from your parent's account to sign up.</p>
            <p>Don't have an access code yet? Grab your nearest parent and ask them to hop on over to the <NavLink to="/parent">Parent Page</NavLink>.
            </p>
            <form>                
                <FormControl sx={{ width: '25ch', m: 2 }}>
                <Stack spacing={2}>
                    <TextField 
                        type="text" 
                        name="username" 
                        value={username} 
                        id="username" 
                        label="Username"
                        onChange={handleChange}>
                    </TextField>

                    <Box>
                        <TextField 
                            type={passwordShowing ? "text" : "password"} 
                            name="password" 
                            value={password} 
                            id="password" 
                            label="Password"
                            onChange={handleChange}>
                        </TextField>
                            {!passwordShowing ?
                            <span className="material-symbols-outlined" onClick={toggle}>visibility</span>
                            :
                            <span className="material-symbols-outlined" onClick={toggle}>visibility_off</span>}
                    </Box>

                    <TextField                
                        type="text" 
                        name="accessCode" 
                        value={accessCode} 
                        id="accessCode"
                        label="Access code" 
                        onChange={handleChange}>
                    </TextField>

                    <Button 
                        id="signup-button" 
                        onClick={handleSubmit}>Submit
                    </Button> 
                </Stack> 
                </FormControl>                           
            </form>
            <p>Just want a quick demo without signing up? Go to <NavLink to="/login">Login</NavLink> and use our demo account.</p>           
        </Paper>
    )
}