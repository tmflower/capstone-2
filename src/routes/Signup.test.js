import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Signup } from './Signup';

const alert = { severity: 'warning', message: 'this is a message'}

// smoke test
test("renders without crashing", () => {
    render(
    <MemoryRouter>
      <Signup alert={alert}/>
    </MemoryRouter>);
  });

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Signup alert={alert}/>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});