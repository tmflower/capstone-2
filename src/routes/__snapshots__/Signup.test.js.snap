// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-17z6wsp-MuiPaper-root"
  >
    <h1>
      Join the Critter Club!
    </h1>
    <p>
      You'll need the access code from your parent's account to sign up.
    </p>
    <p>
      Don't have an access code yet? Grab your nearest parent and ask them to hop on over to the 
      <a
        class=""
        href="/parent"
      >
        Parent Page
      </a>
      .
    </p>
    <form>
      <div
        class="MuiFormControl-root css-16hka9-MuiFormControl-root"
      >
        <div
          class="css-1p5q5e5-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="username"
              id="username-label"
            >
              Username
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="username"
                name="username"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Username
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <span
              class="material-symbols-outlined"
            >
              visibility
            </span>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="accessCode"
              id="accessCode-label"
            >
              Access code
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="accessCode"
                name="accessCode"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Access code
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            id="signup-button"
            tabindex="0"
            type="button"
          >
            Submit
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
